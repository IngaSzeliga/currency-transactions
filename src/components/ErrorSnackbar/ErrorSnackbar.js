import React from "react";
import PropTypes from "prop-types";
import ErrorIcon from "@material-ui/icons/Error";
import CloseIcon from "@material-ui/icons/Close";
import IconButton from "@material-ui/core/IconButton";
import Snackbar from "@material-ui/core/Snackbar";
import SnackbarContent from "@material-ui/core/SnackbarContent";
import "./ErrorSnackbar.scss";

const ErrorSnackbar = props => {
  const { error, handleCloseError } = props;

  return (
    <Snackbar
      className="error-snackbar-container"
      anchorOrigin={{
        vertical: "bottom",
        horizontal: "left"
      }}
      open
      autoHideDuration={6000}
      onClose={handleCloseError}
    >
      <SnackbarContent
        className="error-snackbar-content"
        aria-describedby="client-snackbar"
        message={
          <span id="client-snackbar" className="error-message">
            <ErrorIcon />
            {error}
          </span>
        }
        action={[
          <IconButton
            className="error-icon"
            key="close"
            aria-label="close"
            color="inherit"
            onClick={handleCloseError}
          >
            <CloseIcon />
          </IconButton>
        ]}
      />
    </Snackbar>
  );
};

ErrorSnackbar.propTypes = {
  error: PropTypes.string.isRequired,
  handleCloseError: PropTypes.func.isRequired
};

export default React.memo(ErrorSnackbar);
