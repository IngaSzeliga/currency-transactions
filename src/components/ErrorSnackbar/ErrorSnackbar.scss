@import "../../config/style.scss";

.error-snackbar-container {
  > div:first-child {
    background-color: $dark-red;
    display: flex;
  }
  .error-snackbar-content {
    display: inline-block;

    .error-message {
      justify-content: center;
      display: flex;
      align-items: center;

      > svg:first-child {
        margin-right: 10px;
      }
    }
  }
}
