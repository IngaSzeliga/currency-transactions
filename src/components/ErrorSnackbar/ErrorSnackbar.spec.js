import React from "React";
import { shallow } from "enzyme";
import ErrorSnackbar from ".";

describe("ErrorSnackbar component", () => {
  const component = shallow(
    <ErrorSnackbar error="Some error!" handleCloseError={jest.fn()} />
  );
  it("has working component", () => {
    expect(component).toBeDefined();
  });
  it("expects component to never be changed", () => {
    expect(component).toMatchSnapshot();
  });
});
