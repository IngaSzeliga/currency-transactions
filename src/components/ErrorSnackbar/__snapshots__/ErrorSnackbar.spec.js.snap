// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSnackbar component expects component to never be changed 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  className="error-snackbar-container"
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(SnackbarContent))
    action={
      Array [
        <ForwardRef(WithStyles)
          aria-label="close"
          className="error-icon"
          color="inherit"
          onClick={[MockFunction]}
        >
          <Memo(CloseIcon) />
        </ForwardRef(WithStyles)>,
      ]
    }
    aria-describedby="client-snackbar"
    className="error-snackbar-content"
    message={
      <span
        className="error-message"
        id="client-snackbar"
      >
        <Memo(ErrorIcon) />
        Some error!
      </span>
    }
  />
</WithStyles(ForwardRef(Snackbar))>
`;
